### PR TITLE
[SDL-0190] Fix issue with processing of on_events

### PR DESCRIPTION
**Pull request by LuxoftAKutsan**
Fix https://adc.luxoft.com/jira/browse/FORDTCN-7299 

If a subscription is the last request for resumption, that resumption data processor will revert resumption after it `ClearPendingResumptionRequests` will be called before `SDLPendingResumptionHandler::on_event`, and resumption data processor will think that vehicle data was not subscribed successfully.

Quick fix is reordered subscription to response event. 
SO that pending resumption handler will be subscribed first and redumption data processor the second 

Also renamed ClearPendingResumption to OnResumptionRevert, that is more correct 

Second fix : 
event dispatcher was not support recursive on_event because of shared state. Make variable observers_ local,so it could ro recursive on_event call

### CODE REVIEW

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/sdl_pending_resumption_handler.cc [hunk 1]
```diff
@@ -164,7 +166,7 @@ void SDLPendingResumptionHandler::on_event(
     }
     freezed_resumptions_.clear();
   } else {
-    LOG4CXX_DEBUG(logger_, "Resumption of subscriptions is NOT successful");
+    LOG4CXX_DEBUG(logger_, "Ressubsumption of subscriptions is NOT successful");
```
Comment (AKalinich-Luxoft) [inline src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/sdl_pending_resumption_handler.cc:169]: @LuxoftAKutsan please revert this change

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/sdl_pending_resumption_handler.cc [hunk 2]
```diff
@@ -62,19 +62,21 @@ void SDLPendingResumptionHandler::ClearPendingRequestsMap() {
...
   pending_requests_.clear();
 }
 
-void SDLPendingResumptionHandler::ClearPendingResumptionRequests() {
+void SDLPendingResumptionHandler::OnResumptionRevert() {
   LOG4CXX_AUTO_TRACE(logger_);
   using namespace application_manager;
-  ClearPendingRequestsMap();
-
+  //  ClearPendingRequestsMap();
```
Comment (AKalinich-Luxoft) [inline src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/sdl_pending_resumption_handler.cc:75]: Pls remove this

Comment (LuxoftAKutsan) [reply src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/sdl_pending_resumption_handler.cc]: done

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/sdl_pending_resumption_handler.cc (hunk 1 above)
Comment (LuxoftAKutsan) [reply src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/sdl_pending_resumption_handler.cc]: Fixed

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/sdl_pending_resumption_handler.cc [hunk 3]
```diff
@@ -72,8 +72,11 @@ void SDLPendingResumptionHandler::ClearPendingRequestsMap() {
 void SDLPendingResumptionHandler::OnResumptionRevert() {
   LOG4CXX_AUTO_TRACE(logger_);
   using namespace application_manager;
-  ClearPendingRequestsMap();
 
+  if (!pending_requests_.empty()) {
+    LOG4CXX_DEBUG(logger_, "Still waiting for some resonse");
```
Comment (ychernysheva) [inline src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/sdl_pending_resumption_handler.cc:77]: typo `response`

Comment (LuxoftAKutsan) [reply src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/sdl_pending_resumption_handler.cc:77]: fixed
